File: Baekjoon/Class2/30.10773 제로.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.Deque;

class Main {
    public static void main(String[] args) throws IOException {
        int k;
        long total = 0;
        Deque<Integer> nums = new ArrayDeque<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        k = Integer.parseInt(br.readLine());
        for(int i = 0; i < k; i++) {
            int num = Integer.parseInt(br.readLine());
            if(num == 0) nums.pop(); // 0이면 pop
            else nums.push(num);
        }
        while(!nums.isEmpty())
            total += nums.pop();
        System.out.print(total);
        br.close();
    }
}

File: Baekjoon/Class2/30.10773 제로.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int k, num;
    long long total = 0;
    stack<int> nums;

    cin >> k;
    while (k--) {
        cin >> num;
        if (num == 0) nums.pop(); // 0이면 pop
        else nums.push(num);
    }
    while (!nums.empty()) {
        total += nums.top();
        nums.pop();
    }
    cout << total;
    return 0;
}